### ISSUE TITLE
How to use fit_generator with multiple inputs of different type

### ISSUE THREAD

**Issue by Liang-yc** [keras-team/keras#12861]
In a `Keras` model with the Functional API I need to call `fit_generator` to train on augmented images data using an `ImageDataGenerator`. The problem is that my model has two inputs: image and label. I am just trying to use  `Center Loss` ([another Center Loss repo link](https://github.com/shamangary/Keras-MNIST-center-loss-with-visualization)) and it works fine if I just call `fit` rather than `fit_generator`. Now I want to utilize the image pre-processing methods in `ImageDataGenerator`, and I have to use `fit_generator`. All in all, do you have any examples about this problem?

**Comment by Dref360** [comment]
One way to do it is to do your own `keras.utils.Sequence`, it's pretty easy!

https://dref360.github.io/deterministic-da/

**Comment by Liang-yc** [comment]
> One way to do it is to do your own `keras.utils.Sequence`, it's pretty easy!
> 
> https://dref360.github.io/deterministic-da/

Thanks for your reply. Now I use a cheaper way to solve it.  
```
    def gen_flow_for_two_inputs(gen,X1, y, y1):
        genX1 = gen.flow(X1, y, batch_size=batch_size, seed=666)
        genX2 = gen.flow(X1, y1, batch_size=batch_size, seed=666)
        while True:
            X1i = genX1.__next__()
            X2i = genX2.__next__()
            # Assert arrays are equal - this was for peace of mind, but slows down training
            # np.testing.assert_array_equal(X1i[0],X2i[0])
            yield [X1i[0], X1i[1]], [X1i[1],X2i[1]]
```
Where `gen` is an `ImageDatagenerator`, `X1` is the input image , `y` and `y1` are the label inputs.

**Comment by dynamicwebpaige** [comment]
@fchollet: Would this be useful to add as a feature enhancement to `ImageDataGenerator`?
